garrysmod/lua/includes/extensions/string.lua

```diff
@@ -96,255 +80,215 @@ function string.Explode(separator, str, withpattern)
 	return ret
 end
 
-function string.Split( str, delimiter )
-	return string.Explode( delimiter, str )
+string.Explode = string_Explode
+
+function string.Split( str, delimiter, withpattern )
+	return string_Explode( delimiter, str, withpattern )
 end
 
---[[---------------------------------------------------------
-	Name: Implode( seperator, Table)
-	Desc: Takes a table and turns it into a string
-	Usage: string.Implode( " ", { "This", "Is", "A", "Table" } )
------------------------------------------------------------]]
-function string.Implode( seperator, Table ) return
-	table.concat( Table, seperator )
+function string.Implode( sep, table )
+	return table_concat( table, sep )
 end
 
---[[---------------------------------------------------------
-	Name: GetExtensionFromFilename( path )
-	Desc: Returns extension from path
-	Usage: string.GetExtensionFromFilename("garrysmod/lua/modules/string.lua")
------------------------------------------------------------]]
 function string.GetExtensionFromFilename( path )
-	return path:match( "%.([^%.]+)$" )
+	return string_match( path, "%.([^%.]+)$" ) or ""
 end
 
---[[---------------------------------------------------------
-	Name: StripExtension( path )
------------------------------------------------------------]]
 function string.StripExtension( path )
-	local i = path:match( ".+()%.%w+$" )
-	if ( i ) then return path:sub( 1, i - 1 ) end
-	return path
+	local i = string_match( path, ".+()%.%w+$" )
+	if ( i == nil ) then return path end
+	return string_sub( path, 1, i - 1 )
 end
 
---[[---------------------------------------------------------
-	Name: GetPathFromFilename( path )
-	Desc: Returns path from filepath
-	Usage: string.GetPathFromFilename("garrysmod/lua/modules/string.lua")
------------------------------------------------------------]]
 function string.GetPathFromFilename( path )
-	return path:match( "^(.*[/\\])[^/\\]-$" ) or ""
+	return string_match( path, "^(.*[/\\])[^/\\]-$" ) or ""
 end
 
---[[---------------------------------------------------------
-	Name: GetFileFromFilename( path )
-	Desc: Returns file with extension from path
-	Usage: string.GetFileFromFilename("garrysmod/lua/modules/string.lua")
------------------------------------------------------------]]
 function string.GetFileFromFilename( path )
-	if ( !path:find( "\\" ) && !path:find( "/" ) ) then return path end 
-	return path:match( "[\\/]([^/\\]+)$" ) or ""
+	if ( !string_find( path, "\\", 1, true ) && !string_find( path, "/", 1, true ) ) then return path end
+	return string_match( path, "[\\/]([^/\\]+)$" ) or ""

```

Comment:
What if `return string_match( path, "[\\/]([^/\\]+)$" ) or path`?
And remove line 108